js/iframe-coordinator/elements/frame-router.ts

```diff
@@ -25,6 +42,15 @@ class FrameRouterElement extends HTMLElement {
     window.addEventListener("message", event => {
       this.router.ports.fromClient.send(event.data);
     });
+
+    // Subscribe to component out port to handle messages 
+    this.router.ports.toHost.subscribe(labeledMsg => {
```

Comment:
I changed this to encode and send the entired LabeledMessage object so that I knew the event type.

If we ever add the `protocol` field we discussed today or other meta fields, we might have to revisit this or wrap LabeledMessage with another class internally.

Let me know though if you see issues with exposing this object out this far.